pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -101,58 +100,43 @@ func (r *DNSPolicyReconciler) expectedProbesForGateway(ctx context.Context, gw c
 		log.V(3).Info("DNS Policy has no defined health check")
 		return nil, nil
 	}
+	ipPattern := `\b(?:\d{1,3}\.){3}\d{1,3}\b`
+	re := regexp.MustCompile(ipPattern)
 
-	for _, listener := range gw.Spec.Listeners {
+	for _, address := range gw.Status.Addresses {
+		port := dnsPolicy.Spec.HealthCheck.Port
+		if port == nil {
+			port = &defaultPort
+		}
 
-		log.V(3).Info("getting dnsrecord", "name", dnsRecordName(gw.Name, string(listener.Name)))
-		dnsRecord := &v1alpha1.DNSRecord{}
-		if err := r.Client().Get(ctx, client.ObjectKey{Name: dnsRecordName(gw.Name, string(listener.Name)), Namespace: dnsPolicy.Namespace}, dnsRecord); err != nil {
-			if errors.IsNotFound(err) {
-				continue
-			} else {
-				return nil, err
-			}
+		matches := re.FindAllString(address.Value, -1)
+		if len(matches) != 1 {
+			log.V(3).Info("Found more than 1 ip, skipping")
+			continue
 		}
-		for _, endpoint := range dnsRecord.Spec.Endpoints {
 
-			log.V(1).Info("reconcileHealthChecks: found DNS Record endpoint", "endpoint", endpoint)
-			if endpoint.RecordType == string(v1alpha1.CNAMERecordType) {
-				// if the CNAME is a subdomain of this dnsrecord, skip it
-				if dns.IsSubDomain(dnsRecord.Name, endpoint.Targets[0]) {
-					continue
-				}
-			} else if endpoint.RecordType != string(v1alpha1.ARecordType) {
-				log.V(1).Info("reconcileHealthChecks: not an A record, skipping")
-				continue
-			}
-			for _, target := range endpoint.Targets {
-				port := dnsPolicy.Spec.HealthCheck.Port
-				if port == nil {
-					port = &defaultPort
-				}
-				log.V(1).Info("reconcileHealthChecks: adding health check for target", "target", target)
-				healthCheck := &v1alpha1.DNSHealthCheckProbe{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      fmt.Sprintf("%s-%s", target, dnsRecord.Name),
-						Namespace: gw.Namespace,
-						Labels:    commonDNSRecordLabels(client.ObjectKeyFromObject(gw), client.ObjectKeyFromObject(dnsPolicy)),
-					},
-					Spec: v1alpha1.DNSHealthCheckProbeSpec{
-						Port:                     *port,
-						Host:                     dnsRecord.Name,
-						Address:                  target,
-						Path:                     dnsPolicy.Spec.HealthCheck.Endpoint,
-						Protocol:                 *dnsPolicy.Spec.HealthCheck.Protocol,
-						Interval:                 metav1.Duration{Duration: 60 * time.Second},
-						AdditionalHeadersRef:     dnsPolicy.Spec.HealthCheck.AdditionalHeadersRef,
-						FailureThreshold:         dnsPolicy.Spec.HealthCheck.FailureThreshold,
-						ExpectedResponses:        dnsPolicy.Spec.HealthCheck.ExpectedResponses,
-						AllowInsecureCertificate: dnsPolicy.Spec.HealthCheck.AllowInsecureCertificates,
-					},
-				}
-				healthChecks = append(healthChecks, withDNSRecord(dnsRecord, healthCheck))
```

Comment:
We've updated the expectedProbeforGateways to not depend on the dnsrecord. @philbrookes any objection to getting the dns record still in the function but only using it for this call?

Reply:
I see that now we create the probes by iterating through `.status.addresses` instead of the endpoints. I wonder if there's a way to find the matching listener for each address and set it as an annotation/label of the probe, then it wouldn't be needed to get the DNSRecord at all

Reply:
This should be resolved now. Let me know if there are any further updates needed @sergioifg94

Reply:
Looks good now on that side, thanks @laurafitzgerald I'll mark it as resolved